modules/javafx.base/src/main/java/javafx/collections/ModifiableObservableListBase.java

```diff
@@ -88,14 +89,15 @@ public ModifiableObservableListBase() {
 
     @Override
     public boolean setAll(Collection<? extends E> col) {
+        if (isEmpty() && col.isEmpty()) return false;
         beginChange();
         try {
             clear();
             addAll(col);
+            return true;
```

Comment:
I think following code would be more suitable here, 
```
boolean res = super.addAll(c);
return res;
```
This code is already used in two `addAll()` methods of this class.

Reply:
Makes sense to me. I changed it accordingly.

Reply:
I don't think this change is correct.  `setAll(Collection)` should return true if the list is modified. As discussed in an [earlier comment](#issuecomment-684117392) this means returning true if either the existing Collection or the new Collection is non-empty.

Reply:
I'm sorry, you're absolutely right. I didn't spend enough time thinking about the change and assumed it was fine since the tests passed. Turns out the test didn't cover it. I reverted that part of the change and added a test-case.